src/rf/apps/core/decorators.py

```diff
@@ -43,7 +56,10 @@ def wrapper(request, *args, **kwargs):
                 raise MethodNotAllowed()
 
             if request.method == 'PUT':
-                request.PUT = QueryDict(request.body)
+                request.PUT = PatchedQueryDict(request.PUT.urlencode())
```

Comment:
At this point, `request.PUT` does not yet exist. So `request.PUT.urlencode()` will throw an error.